cnxpublishing/subscribers.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import psycopg2
+from cnxarchive.scripts import export_epub
+from pyramid.events import subscriber
+from pyramid.threadlocal import get_current_registry
+
+
+from . import events
+from .bake import remove_baked, bake
+from .db import (
+    set_post_publications_state,
+    update_module_state,
+    with_db_cursor,
+)
+
+
+logger = logging.getLogger('cnxpublishing')
+
+
+@subscriber(events.PostPublicationEvent)
+@with_db_cursor
+def post_publication_processing(event, cursor):
+    """Process post-publication events coming out of the database."""
+    module_ident, ident_hash = event.module_ident, event.ident_hash
+    logger.debug('Processing module_ident={} ident_hash={}'.format(
+        module_ident, ident_hash))
+    set_post_publications_state(cursor, module_ident, 'Processing')
+    # TODO commit state change
```

Comment:
I was going to ignore this for now, because it's not something the current code does. Though, what you've mentioned here is exactly why I put in the `TODO` item.

> The current post_processing daemon does make stateid == 6 (Processing) visible in the db. 

You're correct, it attempts to do so. Though, I don't see how it would actually update that state, because no commit is actually done until the process completes.
At any rate, the module-state is not updated at the beginning of this function. It's good you spotted this because it's definitely missing, which brings us back to the need to resolve this `TODO`, because otherwise the change to the module-state won't be committed.

> I presume the change in transaction handling is to avoid exposing the deletion of existing baked content until success?

AFAICT there is not change to the transaction handling, other than the update to module-state is definitely missing as a prerequisite task of this procedure.